data/blog/2025/2025-08-06-strangler-fig-pattern.md

```diff
@@ -0,0 +1,139 @@
+
+<!--- EXTENDED -->
+
+In a nutshell, the old platform is put behind an intermediary interface.
+During development, as the new services are coded, they replace old components behind the scenes.
+As the execution is redirected through the new code, the old code can be removed completely until the whole system is migrated to the new architecture.
+
+Below is an image of the Strangle Fig plant which envelops the host tree.
+In some cases it can live alongside the host, while in others, it may cause the host to die.
+
+![strangler-fig](/images/blog/strangler-fig/strangler-fig.jpg "strangler fig")
+
+## When to Use the Strangler Fig Pattern?
+
+**Rewriting the codebase** is always an option, but **is often avoided** in favor of the progressive approach as we will discuss later in this article.
+The recent sunsetting of [**Laminas MVC**](https://docs.laminas.dev/mvc/) provides the perfect example for when to consider migration.
+
+So you have a legacy system like Laminas MVC built with older architecture.
+It was good a decade ago, but has now reached a point where **maintaining it is a chore** that even skilled, well-paid developers shy away from.
+Implementing a new feature is a difficult undertaking which may even be impossible because of the system's limitations when trying to interface with modern dependencies.
+
+The **solution** is obvious when you consider that modern packages work fine with modern platforms and are more easily integrated into them, so this is the moment when the CTO must decide how to proceed.
+If you used Laminas MVC in your platform, the natural progression is toward the Laminas alterative: the [**Mezzio microframework**](https://docs.mezzio.dev/) and its more opinionated implementation, the [**Dotkernel Headless Platform**](https://www.dotkernel.com/).
+
+Migrating between Laminas MVC and Mezzio is like methodically replacing bricks in a wall.
+The wall remains upright, even when a couple of crumbling bricks are being replaced with shiny, new bricks.
+Bricks made of gold (OK, that was a bit much).
+A well-implemented Strangler Fig Pattern will have that wall provide the same functionality throughout its rebuilding process.
+At the end of the migration, the whole wall will contain all-new bricks, built better than before, anxiously awaiting new features.
+
+## Why Not Rewrite the Old Platform Instead?
+
+Many organizations are weighed down with legacy applications that:
+
+- Are a vital business component that can't be put on pause during additional development.
+- Are difficult to change or scale because of platform limitations or sheer complexity.
+- Use outdated or discontinued technology stacks.
+
+One of the options for companies that need to improve their workflow is a **complete rewrite**, but it carries several **disadvantages**:
+
+- There is a risk of not migrating important domain logic and business logic that is not well documented.
```

Comment:
Honestly, this risk also exists with strangler fig.

Reply:
I tried to contrast this by mentioning that the impact for Strangler Fig is minimized, at least, ('Mitigate risks' section) via working with 'small, manageable changes'. There is risk in both approches, but one should come out on top.